### ISSUE TITLE
FSE - Mini-Cart Drawer block does not trigger the recalculation of the gallery size while it removes the scrollbar

### ISSUE THREAD

**Issue by webaxones** [woocommerce/woocommerce-gutenberg-products-block#9635]
Opening the mini cart drawer makes the scrollbar disappear (due to CSS): but the JS does not detect resizing (expansion of the page since the scrollbar width is reduced) so the width of the photo gallery is not recalculated and we see a piece of the next photo.

Since it moves all the elements on all pages of the website, I think adding an overflow-x:hidden to the body would be sufficient when the drawer is open. Of course, this means that one can scroll on the page, but it's not bothersome. At least, it's less bothersome than seeing the elements behind the overlay resize.

Chrome/Firefox
WP 6.2.2
WC 7.7.0
WC Blocks 10.3.0

![mini-cart](https://github.com/woocommerce/woocommerce-blocks/assets/17084006/5046226d-b729-46bf-aca9-65f69a5f6481)
_The above screenshot does not show the entire window by personal choice_

**Comment by gigitux** [comment]
Hi, @webaxones!
Thanks for opening this issue! We added this issue in the main issue about the Mini Cart block (#9198). In the following days, we will take a look at this!

**Comment by webaxones** [comment]
Hi @gigitux 
Thanks,
Here are two short videos to see the generated movements.
![product-page](https://github.com/woocommerce/woocommerce-blocks/assets/17084006/17841ff9-d7a4-4706-be60-3edb40902810)

![prod-category-page](https://github.com/woocommerce/woocommerce-blocks/assets/17084006/2eee0883-ff97-43a7-935a-7ff0f9519d34)

In the meantime, I have added this to my CSS to handle the issue:
```
body.drawer-open {
	overflow: auto !important;
}
```

**Comment by Aljullu** [comment]
Thanks for opening this issue @webaxones!

> Since it moves all the elements on all pages of the website, I think adding an overflow-x:hidden to the body would be sufficient when the drawer is open. Of course, this means that one can scroll on the page, but it's not bothersome. At least, it's less bothersome than seeing the elements behind the overlay resize.

The reason we made the page non-scrollable when the Mini-Cart block was open was because we detected it was confusing and it had a negative impact on accessibility. On mobile devices, it was particularly confusing because the page was completely covered by the drawer and yet it could still be scrolled.

In #9648 I tried an alternative approach. We keep the `<body>` without scroll, but we add a right padding with the same width as the scrollbar. I think that has the benefits of both approaches. :slightly_smiling_face:
